Display help more liberally.

MarkBennett opened the pull request:
Plow now displays help if you:
- don't enter any args or commands
- enter "help" or "usage"

It will continue to display help if you do:

```
plow --help
```

MarkBennett committed:
Display help more liberally.

Plow now displays help if you:

 * don't enter any args or commands
 * enter "help" or "usage"

It will continue to display help if you do:

    plow --help

mm53bar commented on bin/plow:
```diff
@@ -22,6 +22,11 @@ function usage
 
+if test $# = 0; then
+  usage
+  exit
```
Should this exit with a 1 status?

MarkBennett replied:
You're right. Didn't realize exit defaults with an exit code of 1. We're working on it now.

MarkBennett replied:
We just pushed a change to exit with return code 64 if the command is used improperly. This is consistent with other system commands.

mm53bar replied:
`exit 64`, eh?  Haven't heard that one before!

MarkBennett committed:
exit with 0 on help

MarkBennett committed:
use return code EX_USAGE (64) from sysexits.h

Similar to other commands such as mkdir

mm53bar commented on bin/plow:
```diff
@@ -37,9 +42,9 @@ while test $# != 0; do
+    --help|usage|help)
       usage
-      exit
+      exit 0
```
I always assumed that exit 0 was the default and that you had a to force an `exit 1` if that's what you wanted.  Did you guys double check this?

MarkBennett replied:
Using "exit" without setting a value did return 0, however using exit on line 27 returned 1. Rather than relying on developers to understand the nuances of how exit infers it's exit code we figured it was safer to be explicit.

MarkBennett commented:
Is this ready to be merged? Anything else we need to do?

mm53bar commented:
I don't often see `exit 0` so I would prefer to avoid that if possible.  I'm not sure why it exited with a 1 on line 27 but that might be something that needs to be investigated.   

On Tuesday, 9 April, 2013 at 8:33 AM, Mark wrote:

> Is this ready to be merged? Anything else we need to do?
> 
> —
> Reply to this email directly or view it on GitHub (https://github.com/mm53bar/plow/pull/7#issuecomment-16115969).

MarkBennett committed:
Update exit status to follow bash conventions.

Previously, it explicitly returned 0. As this is the default for exit it
confusing and unconventional.

Also added a note explaining the meaning and source of exit 64.

MarkBennett commented:
We just pushed another commit that removes the explicit "exit 0" in favour of just "exit". Also noted where "exit 64" comes from and what it means.

mm53bar commented:
Cool! Thanks for doing that.
